starfyre/file_router.py

```diff
@@ -0,0 +1,63 @@
+from starfyre import create_component, render_root
+from starfyre.exceptions import IndexFileConflictError
+
+import os
+import sys
+from pathlib import Path
+import importlib
+
+
+class FileRouter:
+    """
+    A router that handles file-based routing.
+
+    This router parses the specified pages directory to automatically generate routes based on
+    the file names. Each file in the pages directory is treated as a separate route. 
+
+    Parameters:
+        pages_directory (str): The path to the directory containing the pages files.
+    Example:
+        pages_directory = "test-application/pages"
+        file_router = FileRouter(pages_directory)
+    """
+
+    def __init__(self, pages_directory):
+        self.pages_directory = pages_directory
+
+    def generate_routes(self):
+        """
+        Generate routes and create corresponding HTML files.
```

Comment:
we don't create html files anymore here.

We can possibly name it as `populate_router`?